support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/dao/NoSqlTicketRegistry.java

```diff
@@ -0,0 +1,113 @@
+package org.apereo.cas.dao;
+
+import org.apereo.cas.logout.LogoutManager;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.registry.AbstractTicketRegistry;
+import org.apereo.cas.ticket.registry.TicketRegistryCleaner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.util.Collection;
+
+public class NoSqlTicketRegistry extends AbstractTicketRegistry implements TicketRegistryCleaner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NoSqlTicketRegistry.class);
+    private static final String TICKET_GRANTING_TICKET_PREFIX = "TGT";
+    private static final String SERVICE_TICKET_PREFIX = "ST";
+
+    private NoSqlTicketRegistryDao ticketRegistryDao;
+    private LogoutManager logoutManager;
+
+    @Autowired
+    public NoSqlTicketRegistry(@Qualifier("cassandraDao") final NoSqlTicketRegistryDao ticketRegistryDao, final LogoutManager logoutManager) {
+        this.ticketRegistryDao = ticketRegistryDao;
+        this.logoutManager = logoutManager;
+    }
+
+    @Override
+    public void addTicket(final Ticket ticket) {
+        final String ticketId = ticket.getId();
+        LOGGER.debug("INSERTING TICKET {}", ticketId);
+        if (isTgt(ticketId)) {
+            ticketRegistryDao.addTicketGrantingTicket(ticket);
+        } else if (isSt(ticketId)) {
+            ticketRegistryDao.addServiceTicket(ticket);
+        } else {
+            LOGGER.error("inserting unknown ticket type {}", ticket.getClass().getName());
+        }
+    }
+
+    @Override
+    public int deleteTicket(final String id) {
+        if (deleteSingleTicket(id)) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public boolean deleteSingleTicket(final String id) {
+        LOGGER.debug("DELETING TICKET {}", id);
+        if (isTgt(id)) {
+            return ticketRegistryDao.deleteTicketGrantingTicket(id);
+        } else if (isSt(id)) {
+            return ticketRegistryDao.deleteServiceTicket(id);
+        } else {
+            LOGGER.error("deleting unknown ticket type {}", id);
+            return false;
+        }
+    }
+
+    @Override
+    public Ticket getTicket(final String id) {
+        LOGGER.debug("QUERYING TICKET {}", id);
```

Comment:
In my initial reviews of this branch, I did put together a small example of how this might be simplified via generics and refactoring of the current API. If you'd like I can certainly share that with you to you see what alternatives might exist.